model/src/test/kotlin/AbstractDependencyNavigatorTest.kt

```diff
@@ -0,0 +1,360 @@
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package org.ossreviewtoolkit.model
+
+import io.kotest.core.spec.style.WordSpec
+import io.kotest.matchers.collections.beEmpty
+import io.kotest.matchers.collections.contain
+import io.kotest.matchers.collections.containAll
+import io.kotest.matchers.collections.containExactly
+import io.kotest.matchers.collections.containExactlyInAnyOrder
+import io.kotest.matchers.collections.haveSize
+import io.kotest.matchers.collections.shouldContainExactlyInAnyOrder
+import io.kotest.matchers.should
+import io.kotest.matchers.shouldBe
+import io.kotest.matchers.shouldNot
+
+import java.io.File
+import java.time.Instant
+
+import org.ossreviewtoolkit.utils.test.readOrtResult
+
+/**
+ * A base class for tests of concrete [DependencyNavigator] implementations.
+ *
+ * The class is configured with an ORT result file that contains the expected results in a specific format. It then
+ * runs tests on the [DependencyNavigator] of this result and checks whether it returns the correct dependency
+ * information.
+ */
+abstract class AbstractDependencyNavigatorTest : WordSpec() {
+    /** The name of the result file to be used by all test cases. */
+    protected abstract val resultFileName: String
+
+    /**
+     * The name of the file with a result that contains issues. This is used by tests of the collectIssues() function.
+     */
+    protected abstract val resultWithIssuesFileName: String
+
+    private val testResult by lazy { readOrtResult(resultFileName) }
+
+    private val testProject by lazy { testResult.getProject(PROJECT_ID)!! }
+
+    protected val navigator by lazy { testResult.dependencyNavigator }
+
+    init {
+        "scopeNames" should {
+            "return the scope names of a project" {
+                navigator.scopeNames(testProject) should containExactlyInAnyOrder("compile", "test")
+            }
+        }
+
+        "directDependencies" should {
+            "return the direct dependencies of a project" {
+                navigator.directDependencies(testProject, "test")
+                    .map { it.id }.toList() should containExactly(
+                    Identifier("Maven:org.scalacheck:scalacheck_2.12:1.13.5"),
+                    Identifier("Maven:org.scalatest:scalatest_2.12:3.0.4")
+                )
+            }
+
+            "return an empty sequence for an unknown scope" {
+                navigator.directDependencies(testProject, "unknownScope").toList() should beEmpty()
```

Comment:
Similar to my above comment, this could be

    navigator.directDependencies(testProject, "unknownScope") should beEmptySequence()

with

    import io.kotest.matchers.sequences.beEmpty as beEmptySequence